src/java/org/apache/cassandra/audit/AuditLogManager.java

```diff
@@ -398,4 +418,103 @@ else if (e instanceof PasswordGuardrail.PasswordGuardrailException)
 
         return PasswordObfuscator.obfuscate(e.getMessage());
     }
+
+    private static class JmxFormatter
+    {
+        private static String user(Subject subject)
+        {
+            return String.format("%s", subject == null ? null : subject.getPrincipals().stream().map(Objects::toString).collect(Collectors.joining(", ")));
+        }
+
+        private static String method(Method method, Object[] args)
+        {
+            Function<Object, String> fmt = obj -> {
+                if (obj == null)
+                    return "null";
+                return obj.toString();
+            };
+            String argsFmt = "";
+            if (args != null)
+                argsFmt = Arrays.stream(args).map(fmt).collect(Collectors.joining(", "));
+            return String.format("%s#%s(%s)", method.getDeclaringClass().getCanonicalName(), method.getName(), argsFmt);
+        }
```

Comment:
Rather than embedding `fmt` in the `method` method as a function object like this, I'd move it into its own function and just use a method reference in stead. It avoids creating the function instance each time you format a method for logging:

```suggestion
        private static String method(Method method, Object[] args)
        {
            String argsFmt = "";
            if (args != null)
                argsFmt = Arrays.stream(args).map(JmxFormatter::nullOrToString).collect(Collectors.joining(", "));
            return String.format("%s#%s(%s)", method.getDeclaringClass().getCanonicalName(), method.getName(), argsFmt);
        }

        private static String nullOrToString(Object obj)
        {
            if (obj == null)
                return "null";
            return obj.toString();
        }
```

Reply:
This will reuse the same allocation, but create a new reference on the stack to the anonymous class that the lambda compiles into. Fully agree with the stylistic comment here though, updated.